src/kakarot/interpreter.cairo

```diff
@@ -65,27 +65,28 @@ namespace Interpreter {
         let is_pc_ge_code_len = is_le(evm.message.bytecode_len, pc);
         if (is_pc_ge_code_len != FALSE) {
             let is_precompile = Precompiles.is_precompile(evm.message.code_address.evm);
-            let caller_address = evm.message.caller;
-            // Caller of the contract that is calling the precompile
             if (is_precompile != FALSE) {
                 // If the precompile is called straight from an EOA, the sender_context is the EOA
                 // Otherwise, the sender_context is the caller of the contract that is calling the precompile
                 // This is only relevant for the Kakarot Cairo Module precompile.
                 let parent_context = evm.message.parent;
                 let is_parent_zero = Helpers.is_zero(cast(parent_context, felt));
                 if (is_parent_zero != FALSE) {
-                    tempvar sender_context = evm.message.caller;
+                    // Case A: The precompile is called straight from an EOA
+                    tempvar caller_code_address = evm.message.caller;
```

Comment:
I think it is a little bit confusing as If this is an EOA there is no code
And `caller_code_address` is defined
> The address of the code of the contract that calls the precompile.

Unless I miss something ?

Reply:
also the above comment is not updated and mention sender_context that is no more relevant

Reply:
But in fact after discussion for an EOA the code composant exist but is only empty so not a problem to have `caller_code_address == EOA address`.